expect.go

```diff
@@ -69,7 +69,11 @@ func (c *Console) Expect(opts ...ExpectOpt) (string, error) {
 
 	defer func() {
 		for _, observer := range c.opts.ExpectObservers {
-			observer(matcher, buf.String(), err)
+			if err == nil && matcher != nil {
```

Comment:
It's possible matcher is not nil (we successfully matched) but the callback for `Then` returns err here. I think we should pass `err` along to `observer` and handle that there.

Reply:
Okay, yeah, had the logic wrong.  Now I just look to see if matcher is nil, if it is, then we pass `options.Matchers`, otherwise we pass `[]Matcher{matcher}`.